doc/sphinx/addendum/type-classes.rst

```diff
@@ -146,7 +146,7 @@ the constraints as a binding context before the instance, e.g.:
 
    Program Instance prod_eqb `(EA : EqDec A, EB : EqDec B) : EqDec (A * B) :=
      { eqb x y := match x, y with
-                  | (la, ra), (lb, rb) => andb (eqb la lb) (eqb ra rb)
+                  | (la, lb), (ra, rb) => andb (eqb la ra) (eqb lb rb)
```

Comment:
After some thoughts, this is indeed more readable (`la` and `ra` live in `A` while `lb` and `rb` live in `B`).